handwriting_line_recognition.py

```diff
@@ -0,0 +1,392 @@
+    parser.add_argument("-o", "--lstm_layers", default=1,
+                        help="Number of layers for the LSTM")
+
+    parser.add_argument("-e", "--epochs", default=501,
+                        help="Number of epochs to run")
+    parser.add_argument("-l", "--learning_rate", default=0.0001,
+                        help="Learning rate for training")
+    parser.add_argument("-w", "--lr_scale", default=1,
+                        help="Amount the divide the learning rate")
+    parser.add_argument("-r", "--lr_period", default=30,
+                        help="Divides the learning rate after period")
+
+    parser.add_argument("-s", "--batch_size", default=32,
+                        help="Batch size")
+
+    parser.add_argument("-x", "--random_x_translation", default=0.03,
+                        help="Randomly translation the image in the x direction (+ or -)")
+    parser.add_argument("-y", "--random_y_translation", default=0.03,
+                        help="Randomly translation the image in the y direction (+ or -)")
+    parser.add_argument("-j", "--random_x_scaling", default=0.10,
+                        help="Randomly scale the image in the x direction")
+    parser.add_argument("-k", "--random_y_scaling", default=0.10,
+                        help="Randomly scale the image in the y direction")
+    parser.add_argument("-p", "--random_shearing", default=0.5,
+                        help="Randomly shear the image in radians (+ or -)")
+
+    parser.add_argument("-d", "--log_dir", default="./logs",
+                        help="Directory to store the log files")
+    parser.add_argument("-c", "--checkpoint_dir", default="model_checkpoint",
+                        help="Directory to store the checkpoints")
+    parser.add_argument("-n", "--checkpoint_name", default="handwriting_line.params",
+                        help="Name to store the checkpoints")
+    args = parser.parse_args()
+
+    gpu_id = int(args.gpu_id)
+    ctx = mx.gpu(gpu_id)
+    
+    num_downsamples = int(args.num_downsamples)
+    resnet_layer_id = int(args.resnet_layer_id)
+    lstm_hidden_states = int(args.lstm_hidden_states)
+    lstm_layers = int(args.lstm_layers)
+    
+    epochs = int(args.epochs)
+    learning_rate = float(args.learning_rate)
+    lr_scale = float(args.lr_scale)
+    lr_period = float(args.lr_period)
+    batch_size = int(args.batch_size)
+    
+    random_y_translation, random_x_translation = float(args.random_x_translation), float(args.random_y_translation)
+    random_y_scaling, random_x_scaling = float(args.random_y_scaling), float(args.random_x_scaling)
+    random_shearing = float(args.random_shearing)
+    
+    log_dir = args.log_dir
+    checkpoint_dir, checkpoint_name = args.checkpoint_dir, args.checkpoint_name
+
+    train_ds = IAMDataset("line", output_data="text", train=True)
+    print("Number of training samples: {}".format(len(train_ds)))
+
+    test_ds = IAMDataset("line", output_data="text", train=False)
```

Comment:
make word / line a parameter so that we can switch from one to the other easily, if practical